src/Connection.php

```diff
@@ -99,19 +99,31 @@ public function getSubscriptions()
     private $streamSocket;
 
     /**
+     * Stream wrapper for testing purposes.
+     *
+     * @var mixed StreamWrapper.
+     */
+    private $streamWrapper;
+
+    /**
      * Constructor.
      *
-     * @param ConnectionOptions $options Connection options object.
+     * @param ConnectionOptions $options       Connection options object.
+     * @param StreamWrapper     $streamWrapper Wrapper for stream functions. Testing purposes.
      */
-    public function __construct(ConnectionOptions $options = null)
+    public function __construct(ConnectionOptions $options = null, StreamWrapper $streamWrapper = null)
```

Comment:
Could be possible to not use an $streamWrapper on the constructor?
Just a getter/setter would be fine.